### PR TITLE
CCS data controller scenarios

### PR DESCRIPTION
**Pull request by katstevens**
Trello: https://trello.com/c/BOlmZpkr/341-functional-tests-for-ccs-data-controller-views

- Adds CCS Data Controller role to permissions tests
- Adds CCS Data Controller role login smoke test
- Adds scenarios for the tabbed search box (supplier name/DUNS/companies house)

~Going to hold off merging this until Laurence's `DM_PRODUCTION...` username var changes are sorted, but other feedback on the tests very welcome~ Now done 👍

### CODE REVIEW

File: features/admin/download_user_research_csvs.feature [hunk 1]
```diff
@@ -1,26 +1,27 @@
 @admin @download-user-research-participants
 Feature: Download user research participants
 
-Scenario Outline: Correct users can view the link to download buyer user reseach participants
+Scenario Outline: Correct users can view the link to download buyer user research participants
```
Comment (jonodrew) [inline features/admin/download_user_research_csvs.feature]: Nice catch!

File: config/common.sh [hunk 2]
```diff
@@ -33,6 +33,9 @@ export DM_ADMIN_MANAGER_USER_PASSWORD=${DM_ADMIN_MANAGER_USER_PASSWORD:=Password
 export DM_ADMIN_FRAMEWORK_MANAGER_USER_EMAIL=${DM_ADMIN_FRAMEWORK_MANAGER_USER_EMAIL:=functional-tests-default-admin-framework-manager-user@user.marketplace.team}
 export DM_ADMIN_FRAMEWORK_MANAGER_USER_PASSWORD=${DM_ADMIN_FRAMEWORK_MANAGER_USER_PASSWORD:=Password1234}
 
+export DM_ADMIN_CCS_DATA_CONTROLLER_USER_EMAIL=${DM_ADMIN_CCS_DATA_CONTROLLER_USER_EMAIL:=admin-ccs-data-controller-user@user.marketplace.team}
```
Comment (lfdebrux) [inline config/common.sh]: Are you happy to change the email to be similar to the others? I realise it is a bit verbose...

```suggestion
export DM_ADMIN_CCS_DATA_CONTROLLER_USER_EMAIL=${DM_ADMIN_CCS_DATA_CONTROLLER_USER_EMAIL:=functional-tests-default-admin-ccs-data-controller-user@user.marketplace.team}
```

File: config/common.sh [hunk 3]
```diff
@@ -33,6 +33,9 @@ export DM_ADMIN_MANAGER_USER_PASSWORD=${DM_ADMIN_MANAGER_USER_PASSWORD:=Password
...
+export DM_ADMIN_CCS_DATA_CONTROLLER_USER_PASSWORD=${DM_ADMIN_CCS_DATA_CONTROLLER_USER_PASSWORD:=Password1234}
+
```
Comment (risicle) [inline config/common.sh]: I'm trying to remember what the process is that _creates_ such users...

File: config/common.sh (hunk 2 above)
Comment (katstevens) [reply config/common.sh]: Oh yes, didn't spot that. Will change!

File: config/common.sh (hunk 3 above)
Comment (risicle) [reply config/common.sh]: Ah yup just found where it consumes this env var (with a bit of magic)

File: features/admin/search_for_suppliers.feature [hunk 4]
```diff
@@ -14,6 +15,46 @@ Scenario Outline: Correct users search for a supplier by registered name
...
+Scenario: Admin data controller user can search for a supplier by companies house number
+  Given I am logged in as the existing admin-ccs-data-controller user
+  And I have a supplier with:
+    | name                 | DM Functional Test Supplier - Search companies house number feature |
+    | companiesHouseNumber | 01234567                                                            |
+  And I click the 'View and edit suppliers' link
+  And I click the 'Company registration number' link
+  When I enter '01234567' in the 'Find a supplier by company registration number' field
+  And I click the 'find_supplier_by_company_registration_number_search' button
+  Then I see an entry in the 'Suppliers' table with:
```
Comment (risicle) [inline features/admin/search_for_suppliers.feature]: Fails for me here when run against preview instead giving me:

screenshot_2019-01-31-10-26-57 548

Comment (katstevens) [reply features/admin/search_for_suppliers.feature]: Interesting... will take a look 👀

Comment (katstevens) [reply features/admin/search_for_suppliers.feature]: Now fixed. No idea how that passed locally, must have had some duff data in my local DB.

**Pull request merged**